### ISSUE TITLE
Firewall inbound rule

### ISSUE THREAD

**Issue by SWAN-UA** [MrAlex94/Waterfox#1495]
Waterfox Classic installer (confirmed on latest yesterday) adds inbound firewall rule.
Is it really needed?!
If so would it be better to ask for confirmation of this in installer?
or if not just don't do it.

**Comment by Squall-Leonhart** [comment]
you understand what inbound is, right?

**Comment by TheWhiteAvatar** [comment]
> 
> 
> you understand what inbound is, right?

Inbound means that someone is allowed to connect to your computer. 
So it IS potentially dangerous.

**Comment by Squall-Leonhart** [comment]
> Inbound means that someone is allowed to connect to your computer.

1> For them to pull this off, the software has to be waiting for a connection and the user must know the exact port that its being waited on.
2> behind a NAT even if they know the port and software is waiting, it won't be translated to the pc running that browser instance without a forwarding rule
3> 80 and 8080 must be specifically mapped in a NAT table to a device hosting a http server for an external connection to succeed. - A web browser is not a http server.

Your oversimplification of networking is dangerous

**Comment by TheWhiteAvatar** [comment]
All right, thank you for explaning.
> Your oversimplification of networking is dangerous

I admit that I'm just not educated enough about networking. 

But I always go by the rule: better overestimate a potential danger, than underestimate. :) 
It saved me on several occasions in life.

That's why I wasn't sure about this, until you explained it.

By the way, I have a question: if I would open inbound connections to ports 80 or 8080 for a specific program with Windows Firewall, would it mean that people can only interact with that program specifically? Or they can use those ports to run other stuff on my computer?

**Comment by Squall-Leonhart** [comment]
The ports an application open are opened only for that application, and 80 and 8080 are generally not able to be mapped to anything but http without building your own firmware.

**Comment by TheWhiteAvatar** [comment]
**Squall-Leonhart** got it! Thank you!

**Comment by MrAlex94** [comment]
Nothing has changed in the installer. Could I ask when this happens? What gets added?

**Comment by MrAlex94** [comment]
Please re-open if you can give more details etc, otherwise I assume, depending on settings this is normal.

**Comment by SWAN-UA** [comment]
Adding Inbound firewall rule - IS NOT A NORMAL!
Especially if there is no explanation why and done without asking for user's permission.
And there is no "reopen" button.

**Comment by grahamperrin** [comment]
> installer

Was it part of the integrated update routine, from one version of Waterfox Classic to another?

**Comment by SWAN-UA** [comment]
Do you mean it isn't reproducible and you don't see that?!
I don't use "integrated" update and service, just install with upgrade, i.e. manual download and install over already installed.
and it was in previous release at least as well, i.e. it isn't appeared exactly in "latest" release.

**Comment by SWAN-UA** [comment]
I have verified this on another environment and with "Waterfox" (not Classic) it is 100% reproducible.

**Comment by TheWhiteAvatar** [comment]
I found information like this: 
[https://superuser.com/questions/882494/why-does-firefox-create-inbound-firewall-rules](https://superuser.com/questions/882494/why-does-firefox-create-inbound-firewall-rules)

Apparently such a behavior was introduced by Firefox several years ago, and even after they discontinued their "Hello" service they left those rules in place. 

Seems like Waterfox simply does the same thing as Firefox now.

What I don't like about those rules (I found them on my computer as well) is that they aren't for ports like 80 / 8080, but for ALL ports and IPs, which should allow anyone to use possible vulnerabilities in Waterfox to harm a computer.
